Tests/SwiftLintFramework/ReporterTests.swift

```diff
@@ -91,4 +91,15 @@ class ReporterTests: XCTestCase {
                 "\t</testcase>\n</testsuite></testsuites>"
         )
     }
+
+    func testHTMLReporter() {
+        let generatedHTML = HTMLReporter.generateReport(generateViolations())
+
+        let formatter = NSDateFormatter()
+        formatter.dateStyle = .ShortStyle
+        let dateString = formatter.stringFromDate(NSDate())
+
+        let originalHTML = "<!doctype html><html><head><title>Swiftlint Report</title><style type='text/css'>table { border: 1px solid gray; border-collapse: collapse; -moz-box-shadow: 3px 3px 4px #AAA; -webkit-box-shadow: 3px 3px 4px #AAA; box-shadow: 3px 3px 4px #AAA; } td, th { border: 1px solid #D3D3D3; padding: 5px 10px 5px 10px; } th { border-bottom: 1px solid gray; background-color: #29345C50; } .error, .warning {background-color: #f0f099;} .error{ color: #ff0000;} .warning { color: #b36b00;}</style></head><body><h1>Swiftlint Report</h1><hr /><h2>Violations</h2><table border=\"1\" style=\"vertical-align: top; height: 64px;\"><thead><tr><th style=\"width: 60pt;\"><b>Serial No.</b></th><th style=\"width: 500pt;\"><b>File</b></th><th style=\"width: 60pt;\"><b>Location</b></th><th style=\"width: 60pt;\"><b>Severity</b></th><th style=\"width: 500pt;\"><b>Message</b></th></tr></thead><tbody><tr><td align=\"right\">1</td><td>filename</td><td align=\"center\">1:2</td><td class='warning'>Warning </td><td>Violation Reason.</td></tr><tr><td align=\"right\">2</td><td>filename</td><td align=\"center\">1:2</td><td class='error'>Error </td><td>Violation Reason.</td></tr></tbody></table><br/><h2>Summary</h2><table border=\"1\" style=\"vertical-align: top; height: 64px;\"><tbody><tr><td>Total files with violations</td><td>1</td></tr><tr><td>Total warnings</td><td>1</td></tr><tr><td>Total errors</td><td>1</td></tr></tbody></table><hr /><p>Created with <a href=\"https://github.com/realm/SwiftLint\"><b>Swiftlint</b></a> 0.12.0 on: \(dateString)</p></body></html>"// swiftlint:disable:this line_length
```

Comment:
As I tested, generated report is a text file that contains 11422 violations in one line.
I don't have tools that can handle such text file well without formatting before open it.